cross compile

This patch is ported from WindRiver linux and to fix cross compile
failure.

And original commits are:
commit 8d5ccc8113e1b51b0529a00c18a4aba956247e1b
commit 5c4212084b871a0c0fb7d174280ec9a634637deb

Upstream-Status: Pending

Signed-off-by: Kang Kai <kai.kang@windriver.com>
---
 bin/mklib |   24 ++++++++++++------------
 1 file changed, 12 insertions(+), 12 deletions(-)

diff --git a/bin/mklib b/bin/mklib
index 9bac29e..b33aa90 100755
--- a/bin/mklib
+++ b/bin/mklib
@@ -49,8 +49,8 @@ expand_archives() {
                     /*) ;;
                     *)  FILE="$ORIG_DIR/$FILE" ;;
                 esac
-                MEMBERS=`ar t $FILE`
-                ar x $FILE
+                MEMBERS=`${AR} t $FILE`
+                ${AR} x $FILE
                 for MEMBER in $MEMBERS ; do
                     NEWFILES="$NEWFILES $DIR/$MEMBER"
                 done
@@ -77,7 +77,7 @@ expand_archives() {
 make_ar_static_lib() {
     OPTS=$1
     shift;
-    RANLIB=$1
+    USE_RANLIB=$1
     shift;
     LIBNAME=$1
     shift;
@@ -87,11 +87,11 @@ make_ar_static_lib() {
     rm -f ${LIBNAME}
 
     # make static lib
-    ar ${OPTS} ${LIBNAME} ${OBJECTS}
+    ${AR} ${OPTS} ${LIBNAME} ${OBJECTS}
 
     # run ranlib
-    if [ ${RANLIB} = 1 ] ; then
-        ranlib ${LIBNAME}
+    if [ ${USE_RANLIB} = 1 ] ; then
+        ${RANLIB} ${LIBNAME}
     fi
 
     echo ${LIBNAME}
@@ -313,9 +313,9 @@ case $ARCH in
 	if [ "x$LINK" = "x" ] ; then
 	    # -linker was not specified so set default link command now
             if [ $CPLUSPLUS = 1 ] ; then
-                LINK=g++
+                LINK=$CXX
             else
-                LINK=gcc
+                LINK=$CC
             fi
 	fi
 
@@ -535,9 +535,9 @@ case $ARCH in
 	if [ "x$LINK" = "x" ] ; then
 	    # -linker was not specified so set default link command now
             if [ $CPLUSPLUS = 1 ] ; then
-                LINK=g++
+                LINK=$CXX
             else
-                LINK=gcc
+                LINK=$CC
             fi
 	fi
 
@@ -903,9 +903,9 @@ case $ARCH in
 	if [ "x$LINK" = "x" ] ; then
 	    # -linker was not specified so set default link command now
             if [ $CPLUSPLUS = 1 ] ; then
-                LINK=g++
+                LINK=${CXX}
             else
-                LINK=gcc
+                LINK=${CC}
             fi
 	fi
 
-- 
1.7.10.4

